src/Tribe/Integrations/WPML/Meta.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * Translate post ids in Event meta data.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Integrations__WPML__Meta {
+
+	/**
+	 * @since TBD
+	 *
+	 * @var Tribe__Events__Integrations__WPML__Meta
+	 */
+	protected static $instance;
+
+	/**
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Integrations__WPML__Meta
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Translates post id in the Event meta data.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return string The translated id for _EventOrganizerID & _EventVenueID.
+	 */
+	public function translate_post_id( $value, $object_id, $meta_key ) {
+
+		if ( isset( $_POST ) ) {
+			return;
+		}
+
+		$accepted_values = array( '_EventOrganizerID', '_EventOrganizerID_Order', '_EventVenueID' );
+
+		if ( ! in_array( $meta_key, $accepted_values ) ) {
+			return $value;
+		}
+
+		$value = $this->get_post_meta( $object_id, $meta_key );
+
+		if ( empty( $value ) ) {
+			return false;
+		}
+
+		$type = strpos( 'Organizer', $meta_key )
+			? Tribe__Events__Organizer::POSTTYPE
+			: Tribe__Events__Venue::POSTTYPE;
+
+		foreach ( $value as & $post_id ) {
+			if ( is_serialized( $post_id ) ) {
+				$array = (array) unserialize( $post_id );
+				foreach ( $array as & $id ) {
+					$id = apply_filters( 'wpml_object_id', $id, $type, true );
```

Comment:
We should have some sort of docblocks pointing to their documentation, besides the normal stuff we use for our internal docblocks.